### PR TITLE
Jevans/user status backup

### PR DESCRIPTION
**Pull request by James-revans**
User status settings in place with api functionality ready for real endpoints.

### CODE REVIEW

File: src/api/types/reactn.model.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export type ReactNStateReducer<StateType, PayloadType> = (
```
Comment (m4ttheweric) [inline src/api/types/reactn.model.ts]: This already exists in ./src/types/reactn.model.ts
Maybe update your import and nix this file?

File: src/api/types/user-status.model.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { AppRoute } from '../../navigation/app-routes';
+
```
Comment (m4ttheweric) [inline src/api/types/user-status.model.ts]: I'm not sure this file belongs in the api directory.

File: src/data/sites.ts [hunk 3]
```diff
@@ -11,6 +11,11 @@ export const Sites: Array<ISite> = [
       url: 'https://api.mbmd.dev',
       apiPath: '/api/1.0',
       appMoniker: 'mbmd'
+      // id: 'm2dev',
```
Comment (m4ttheweric) [inline src/data/sites.ts]: probably shouldn't commit changes to this file, but no biggy.

File: src/native/device-dimensions.state.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ interface IDeviceDimensionsState {
    screen: ScaledSize;
    isLargeScreen: boolean;
    isSmallScreen: boolean;
+   large: 650;
```
Comment (m4ttheweric) [inline src/native/device-dimensions.state.ts]: Not sure what this is doing? Can you explain?

File: src/navigation/home.navigator.tsx [hunk 5]
```diff
@@ -22,7 +22,8 @@ const UserPages = [
    'readerboard',
    'calendar',
    'phonelist',
-   'formTest'
+   'formTest',
+   'userStatus'
```
Comment (m4ttheweric) [inline src/navigation/home.navigator.tsx]: So, I actually would disagree with putting userStatus in the UserPages array. I think this page should be visible to everyone no matter what, and hard coding it into the All Pages view is probably the right approach.

File: src/scenes/callboard/callboard-info.component.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ interface ICallboardInfo {
 
 // Default props defined below function declaration
 
-export const CallboardInfo: React.FC<ICallboardInfo> = props => {
+export const CallboardInfo: React.FC<ICallboardInfo> = (props, theme) => {
```
Comment (m4ttheweric) [inline src/scenes/callboard/callboard-info.component.tsx]: Don't type a component. As we discussed.

File: src/scenes/patients/patients.component.tsx [hunk 7]
```diff
@@ -19,6 +19,17 @@ import { useNavigation } from '@react-navigation/native';
...
+//    patient.ADMIT_DATE = new Date(patient.ADMIT_DATE).toLocaleDateString();
+//    patient.DISCHARGE_DATE = new Date(
+//       patient.DISCHARGE_DATE
+//    ).toLocaleDateString();
+//    patient.LAST_VIEW_DATE = new Date(
+//       patient.LAST_VIEW_DATE
+//    ).toLocaleDateString();
+// });
+
 export const PatientsScreen = ({ route }): SafeAreaLayoutElement => {
```
Comment (m4ttheweric) [inline src/scenes/patients/patients.component.tsx]: Please go through and remove types from components.

File: src/utils/debounce.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React, { useState, useEffect } from 'react';
+
+export function useDebounce(watchValue, delay) {
```
Comment (m4ttheweric) [inline src/utils/debounce.ts]: unused?

File: src/api/types/reactn.model.ts (hunk 1 above)
Comment (James-revans) [reply src/api/types/reactn.model.ts]: File deleted and imports updated

File: src/api/types/user-status.model.ts (hunk 2 above)
Comment (James-revans) [reply src/api/types/user-status.model.ts]: Moved to types folder outside of api directory.

File: src/native/device-dimensions.state.ts (hunk 4 above)
Comment (James-revans) [reply src/native/device-dimensions.state.ts]: I needed the value of 650 in one of my components as opposed to the boolean I must have added this when I didn't understand providers and interfaces as thoroughly. I've removed it from this interface and instead exported the `LARGE_SCREEN_MIN_WIDTH` value to wherever I need the value of 650.

File: src/navigation/home.navigator.tsx (hunk 5 above)
Comment (James-revans) [reply src/navigation/home.navigator.tsx]: userStatus has been removed from this array and has been hard coded to the All Pages view.

File: src/scenes/patients/patients.component.tsx (hunk 7 above)
Comment (James-revans) [reply src/scenes/patients/patients.component.tsx]: All types removed from components

File: src/utils/debounce.ts (hunk 8 above)
Comment (James-revans) [reply src/utils/debounce.ts]: File removed

**Pull request merged**